packages/camera/camera_windows/windows/camera_plugin.cpp

```diff
@@ -0,0 +1,593 @@
+
+const char kCameraNameKey[] = "cameraName";
+const char kResolutionPresetKey[] = "resolutionPreset";
+const char kEnableAudioKey[] = "enableAudio";
+
+const char kCameraIdKey[] = "cameraId";
+const char kMaxVideoDurationKey[] = "maxVideoDuration";
+
+const char kResolutionPresetValueLow[] = "low";
+const char kResolutionPresetValueMedium[] = "medium";
+const char kResolutionPresetValueHigh[] = "high";
+const char kResolutionPresetValueVeryHigh[] = "veryHigh";
+const char kResolutionPresetValueUltraHigh[] = "ultraHigh";
+const char kResolutionPresetValueMax[] = "max";
+
+const std::string kPictureCaptureExtension = "jpeg";
+const std::string kVideoCaptureExtension = "mp4";
+
+// Looks for |key| in |map|, returning the associated value if it is present, or
+// a nullptr if not.
+const EncodableValue *ValueOrNull(const EncodableMap &map, const char *key) {
+  auto it = map.find(EncodableValue(key));
+  if (it == map.end()) {
+    return nullptr;
+  }
+  return &(it->second);
+}
+
+// Parses resolution preset argument to enum value
+ResolutionPreset ParseResolutionPreset(const std::string &resolution_preset) {
+  if (resolution_preset.compare(kResolutionPresetValueLow) == 0) {
+    return ResolutionPreset::RESOLUTION_PRESET_LOW;
+  } else if (resolution_preset.compare(kResolutionPresetValueMedium) == 0) {
+    return ResolutionPreset::RESOLUTION_PRESET_MEDIUM;
+  } else if (resolution_preset.compare(kResolutionPresetValueHigh) == 0) {
+    return ResolutionPreset::RESOLUTION_PRESET_HIGH;
+  } else if (resolution_preset.compare(kResolutionPresetValueVeryHigh) == 0) {
+    return ResolutionPreset::RESOLUTION_PRESET_VERY_HIGH;
+  } else if (resolution_preset.compare(kResolutionPresetValueUltraHigh) == 0) {
+    return ResolutionPreset::RESOLUTION_PRESET_ULTRA_HIGH;
+  } else if (resolution_preset.compare(kResolutionPresetValueMax) == 0) {
+    return ResolutionPreset::RESOLUTION_PRESET_MAX;
+  }
+  return ResolutionPreset::RESOLUTION_PRESET_AUTO;
+}
+
+bool HasCurrentTextureId(int64_t current_camera_id, const EncodableMap &args) {
+  const auto *camera_id =
+      std::get_if<std::int64_t>(ValueOrNull(args, kCameraIdKey));
+
+  if (!camera_id) {
+    return false;
+  }
+  return current_camera_id == *camera_id;
+}
+
+std::unique_ptr<CaptureDeviceInfo> GetDeviceInfo(IMFActivate *device) {
+  assert(device);
+  auto device_info = std::make_unique<CaptureDeviceInfo>();
+  wchar_t *name;
+  UINT32 name_size;
+
+  HRESULT hr = device->GetAllocatedString(MF_DEVSOURCE_ATTRIBUTE_FRIENDLY_NAME,
+                                          &name, &name_size);
+  if (SUCCEEDED(hr)) {
+    wchar_t *id;
+    UINT32 id_size;
+    hr = device->GetAllocatedString(
+        MF_DEVSOURCE_ATTRIBUTE_SOURCE_TYPE_VIDCAP_SYMBOLIC_LINK, &id, &id_size);
+
+    if (SUCCEEDED(hr)) {
+      device_info->display_name = Utf8FromUtf16(std::wstring(name, name_size));
+      device_info->device_id = Utf8FromUtf16(std::wstring(id, id_size));
+    }
+
+    ::CoTaskMemFree(id);
```

Comment:
Please replace manual memory management with smart pointers whenever possible; there's a fair amount of non-trivial logic around memory in this code. I believe `CComHeapPtr` will cover many of the cases.

Reply:
I added CComHeapPtr from atlbase.h and these are working great on local environment as I have ATL support installed. 
But as ATL support is not available on test environment, build fails: 
`camera_plugin.cpp(7,10): fatal error C1083: Cannot open include file: 'atlbase.h': No such file or directory`
Requiring ATL support for developers might not be a good idea either.

Reply:
Sorry, I forgot that it was from ATL. A custom `unique_ptr` sounds like a good option.